pkg/parser/document_processing_apply_substitutions.go

```diff
@@ -74,6 +76,37 @@ func applyAttributeSubstitutions(elements []interface{}, attrs types.AttributesW
 
 }
 
+func applyCounterSubstitution(c types.CounterSubstitution, attrs types.AttributesWithOverrides) (interface{}, bool, error) {
+	counter := attrs.Counters[c.Name]
```

Comment:
should you do  something like 
```
counter, found := attrs.Counters[c.Name]
if !found {
    counter = 0
}
```

Reply:
The default value for the counter will be zero anyway.  So that's fine.

Reply:
Well, default will be *nil* is what I meant to say.  I just didn't see a reason to introduce another variable.

Reply:
sure, but my point was that `counter, found := attrs.Counters[c.Name]` is more idiomatic (I believe)

Reply:
I don't think that's necessarily more idiomatic -- unless you want to capture the case where a nil value might have been stored in the map vs. not being present in the table at all.

I think the idiomatic approach is to have zero values be reasonable.

Having said that, this is mostly down to a matter of taste.  If you want me to make the change I will.